misc/python/materialize/version_list.py

```diff
@@ -222,73 +220,3 @@ def get_previous_mz_version(
         and v not in excluded_versions
     ]
     return max(all_suitable_previous_versions)
-
-
-class VersionList:
-    def __init__(self) -> None:
-        self.versions: list[MzVersion]
-        assert False
-
-    def all_versions(self) -> list[MzVersion]:
-        return self.versions
-
-    def minor_versions(self) -> list[MzVersion]:
-        """Return the latest patch version for every minor version."""
-        minor_versions = {}
-        for version in self.versions:
-            minor_versions[f"{version.major}.{version.minor}"] = version
-
-        assert len(minor_versions) > 0
-        return sorted(minor_versions.values())
-
-    def patch_versions(self, minor_version: MzVersion) -> list[MzVersion]:
-        """Return all patch versions within the given minor version."""
-        patch_versions = []
-        for version in self.versions:
-            if (
-                version.major == minor_version.major
-                and version.minor == minor_version.minor
-            ):
-                patch_versions.append(version)
-
-        assert len(patch_versions) > 0
-        return sorted(patch_versions)
-
-
-class VersionsFromDocs(VersionList):
-    """Materialize versions as listed in doc/user/content/versions
-
-    Only versions that declare `versiond: true` in their
-    frontmatter are considered.
-
-    >>> len(VersionsFromDocs().all_versions()) > 0
-    True
-
-    >>> len(VersionsFromDocs().minor_versions()) > 0
-    True
-
-    >>> len(VersionsFromDocs().patch_versions(minor_version=MzVersion.parse_mz("v0.52.0")))
-    4
-
-    >>> min(VersionsFromDocs().all_versions())
-    MzVersion(major=0, minor=27, patch=0, prerelease=None, build=None)
-    """
-
-    def __init__(self) -> None:
-        files = Path(MZ_ROOT / "doc" / "user" / "content" / "releases").glob("v*.md")
-        self.versions = []
-        for f in files:
-            base = f.stem
-            metadata = frontmatter.load(f)
-            if not metadata.get("released", False):
```

Comment:
Did we exclude not released versions here on purpose? Was the goal to skip older not released versions or to wait for the (latest) version to reach production?

Reply:
Ok, thanks. I understand that it is okay to change this behavior?